zsl/resource/model_resource.py

```diff
@@ -83,11 +88,18 @@ class ResourceQueryContext(object):
     """
 
     def __init__(self, params, args, data):
-        self._params = params
+        # type: (dict, list, dict) -> ()
         self._args = args.copy()
         self._args_original = args
         self._data = data
 
+        # test if params is a list
+        try:
+            len(params)
```

Comment:
why not write it in a way `if not isistance(params, list)` or better using this http://stackoverflow.com/questions/1952464/in-python-how-do-i-determine-if-an-object-is-iterable

by Clean Code using comments to explain strange things is not recommended. At least we could extract a method which does it by `is_listable` and use this. the name would explain what is done there.